### PR TITLE
Use Theano's cuDNN batch normalization for training

### PR DESCRIPTION
**Pull request by gvtulder**
The Theano backend currently uses the cuDNN batch normalization for inference (`theano.sandbox.cuda.dnn.dnn_batch_normalization_test`), but does not yet use the training function `theano.sandbox.cuda.dnn.dnn_batch_normalization_train`. This training function directly computes the means and variances. Instead, keras.backend's `normalize_batch_in_training` computes the variance and mean on its own.

This patch would let `normalize_batch_in_training` use the cuDNN training function, similar to what `batch_normalization` already does.

A simple benchmark suggests that using the cuDNN BN training function can be much faster:
```
Without dnn_batch_normalization_train:
    Time per iteration: 435.557 ms
With dnn_batch_normalization_train:
    Time per iteration: 357.126 ms
```
To check the implementation I computed the differences between the normalized variables, the variances and means computed by the old implementations. Apart from small rounding errors the results were the same.

----

Code used to run the benchmarks:
```python
import time
import numpy as np

import keras.layers as kl
import keras.models as km

input_shape = (64, 128, 128)
batch_size = 32

print 'Constructing model'
inputs = kl.Input(shape=input_shape)
x = kl.BatchNormalization(mode=0, axis=1)(inputs)
x = kl.Flatten()(x)
outputs = kl.Dense(1, activation='sigmoid')(x)

model = km.Model(input=inputs, output=outputs)
model.compile(optimizer='sgd',
              loss='binary_crossentropy',
              metrics=['accuracy'])

print 'Generating data'
data = 5 + 30 * np.random.randn(batch_size, *input_shape)
labels = np.random.rand(batch_size, 1)

print 'Warmup'
model.train_on_batch(data, labels)

print 'Measuring'
t0 = time.time()
n = 0
while n < 100:
    tstart = time.time()
    model.train_on_batch(data, labels)
    tend = time.time()
    print "Iteration: %d train on batch time: %7.3f ms." %( n, (tend - tstart)*1000 )
    n += 1
t1 = time.time()

print "Image shape:", input_shape
print "Batch size: %d" % batch_size
print "Iterations: %d" % n
print "Time per iteration: %7.3f ms" % ((t1 - t0) * 1000 / n)
```

### CODE REVIEW

**Comment by fchollet** [comment]
LGTM, thanks
